dependency/engine.go

```diff
@@ -397,9 +397,14 @@ func (engine *Engine) requestStart(name string, delay time.Duration) {
 	// which should make bugs more obvious
 	if delay > time.Duration(0) {
 		if engine.config.BackoffFactor > 0 {
-			delay = time.Duration(float64(delay) * math.Pow(engine.config.BackoffFactor, float64(info.recentErrors-1)))
-			if engine.config.MaxDelay > 0 && delay > engine.config.MaxDelay {
+			// Use the float64 values for max comparison. Otherwise when casting
+			// the float back to a duration we hit the int64 max which is negative.
+			maxDelay := float64(engine.config.MaxDelay)
+			floatDelay := float64(delay) * math.Pow(engine.config.BackoffFactor, float64(info.recentErrors-1))
+			if engine.config.MaxDelay > 0 && floatDelay > maxDelay {
```

Comment:
I'd use `maxDelay > 0` here, just seems simpler to think about.

Reply:
Fair call. Updated.

Reply:
maybe we want to rethink the value here. because float64 only supports 2^1024 as well. And while 120 was a lot, it wasn't that far.
We could do:
log(MaxDelay)/log(BackoffFactor) to work out the max number of bounces to hit max and then just do
if retries > maxBounces && maxDelay > 0 {
  delay := maxDelay
}

thoughts?
(there is a bit of math to get right, but my point is to deal with retry count rather than an exponentially growing  number)

Reply:
I suppose that would be slightly cleaner than dealing with +Inf from the result of the float calculations.